path_login.go

```diff
@@ -361,6 +360,28 @@ func (b *jwtAuthBackend) createIdentity(allClaims map[string]interface{}, role *
 	return alias, groupAliases, nil
 }
 
+// Checks if there's a custom provider_config and calls FetchGroups() if implemented
+func (b *jwtAuthBackend) fetchGroups(allClaims map[string]interface{}, role *jwtRole) (interface{}, error) {
+	pConfig, err := NewProviderConfig(b.cachedConfig, ProviderMap())
```

Comment:
I'm seeing that the custom provider `Initialize()` method within `NewProviderConfig()` will end up being called upon each login. I think that's okay given that the azure provider doesn't do much there. I'm wondering if we would still want that to happen if there is a custom provider with more or potentially slow initialization code. It appears that the customer provider is already initialized when the backend gets its config.

Not a blocker. Just wanted to see what your thoughts were :)

Reply:
Good question, I guess I envisioned the Initialize() method would mainly be used to translate the "provider_config" values into the CustomProvider struct, and not actually run any API calls against a provider (azure in this case). But it's something to keep in mind as this evolves for sure.